core/src/main/scala-3/chisel3/Module.scala

```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3
+
+import chisel3.experimental.{BaseModule, SourceInfo}
+
+object Module extends ObjectModuleImpl with SourceInfoDoc {
+
+  /** A wrapper method that all Module instantiations must be wrapped in
+    * (necessary to help Chisel track internal state).
+    *
+    * @param bc the Module being created
+    *
+    * @return the input module `m` with Chisel metadata properly set
+    */
+  // TODO(adkian-sifive) the callsite here explicitly passes
+  // sourceInfo so it cannot be a contextual parameter
+  def apply[T <: BaseModule](bc:    => T): T = _applyImpl(bc)
+  def do_apply[T <: BaseModule](bc: => T)(implicit sourceInfo: SourceInfo): T = apply(bc)
```

Comment:
```suggestion
  def apply[T <: BaseModule](bc:    => T)(using sourceInfo: SourceInfo): T = _applyImpl(bc)
```
As mentioned above, change that `Module.do_apply` call in `chisel/core/src/main/scala/chisel3/internal/package.scala` to use `_applyImpl`.

Reply:
so making this change + changing the `Module.do_apply` to `Module._applyImpl` in internal/package.scala throws this error
```
core/src/main/scala/chisel3/internal/package.scala:133:12: method _applyImpl in trait ObjectModuleImpl cannot be accessed as a member of object chisel3.Module from package object internal in package internal
[error]  Access to protected method _applyImpl not permitted because
[error]  enclosing package object internal in package internal is not a subclass of
[error]  trait ObjectModuleImpl in package chisel3 where target is defined
[error]     Module._applyImpl(new ViewParentAPI)(UnlocatableSourceInfo)
[error]            ^
```

Reply:
not really sure what subclassing requirement is not being satisfied here

Reply:
i'm guessing it's cause _applyImpl is protected in ObjectModuleImpl?

Reply:
Right, try making it `protected[chisel3]` instead